[RDY] Fixes hammer of the hunter, island temple basement (Arcana mod)

DangerNoodle opened the pull request:
Fixed hammer of the hunter using charges to meditate (unintended consequence of https://github.com/CleverRaven/Cataclysm-DDA/pull/18225), fixed flooring inconsistencies in island temple basement (one section had wooden floors underneath furniture instead of rock floor).

Additionally fixed certain item spawns not occurring, and mitigated the issue of archons phasing through the floor to chase monsters they should be allied with (via same faction workaround as with the moruboru).

DangerNoodle commented:
Anomaly: boss enemy in the float temple can get outside if z-levels are enabled, even phasing through the floor or ceiling (or otherwise somehow bypassing doors).

Any suggestions to prevent this?

BorkBorkGoesTheCode commented:
I think that is a known issue with z levels. No fixes have been adopted.

DangerNoodle commented:
I have also noticed the possibility that bosses like the archon might be targeting nether monsters. I thought the cultist faction ignored nether monsters?

DangerNoodle commented:
And furthermore, it appears that the fixed, one-hundred-percent spawns do not always appear. Any suggestions?

mugling commented:
Anyone play Arcana and can confirm this works as intended?

DangerNoodle commented:
My testing so far seems to indicate it, but I need to ensure the archon does not get itself killed by phasing through the floor and attacking mi-gos. This does not seem to occur with any reliability, hence testing the archon is a pain.

Further, suddenly there is an astyle regression?

DangerNoodle commented:
Jenkins message does not help, there are no changes to source files in this pull request, and no actual changed files are cited by the messages.

Is there something I am missing, or has the error-checker bugged out in a fun new way?

mugling commented:
Project-wide breakage from #18351, is now resolved

DangerNoodle commented:
I see, peculiar. Glad that it is resolved.

chaosvolt commented:
Because I'm the only one that still pays attention to my old mods, bothered to grab the branch and mess with it. Sorry for the fuckup regarding how I suggested fixing the hammer. >.<

Flying polyps seem to be prone to killing the archon, but at least he stayed in the temple itself. Dementias smashed the place up though, might be better to add them to the surface spawn and edit them out of the upper level

Also, one possible cause for the archon escaping the temple, the rock floors don't have a roof. Doh.

Also noticed a change you didn't mention in the summary, making the dracolich not smash its way outta the boss room. I don't remember why I added that flag, but seems fine to remove it. That's how my turn in Utterly Mad's succession game ended, zombie dragons do not play well with having z-levels enabled.

Instead, only other screwups encountered were unrelated errors, someone removed diamond weapons without fixing the material art weapon lists. :V

DangerNoodle commented:
Added roof section for floating temple, removed monsters from the third floor. I checked the factions and there should be no cause for infighting between the archon and other enemies.

If that is insufficient, it may take nothing short of making the archon deaf to avoid premature death. This may require it generate light or have night vision, to make up for reduced ability to sense the player.

Additionally, apologies for having yet to experiment with the mapgen additions you linked me to some time ago. I will test it and implement in a separate pull request, as it would be better to have these fixes merged without additional content slowing it down.

DangerNoodle commented:
Testing has revealed that the load error cited by Chaosvolt is related, material arts edits were neglected.

DangerNoodle commented:
Is there a reason why it would be creating a merge conflict? Was there a recent linting pull request for this mod? If not, then I may be unable to resolve this conflict.

BorkBorkGoesTheCode commented:
Yes. #18665 and the CVD machine rework

DangerNoodle commented:
I see now what the issue is. I accidentally introduced the error into my test build by copying the mod folder to test it, and mistakenly assumed it was due to mod neglect.

I will revert this commit, use of copy-from can be added in a separate pull request.

DangerNoodle commented:
It should be ready now.

DangerNoodle commented:
Anything else this requires?

chaosvolt commented:
@Coolthulhu likely missed this on his last sweep of merging ready-to-go PRs.

DangerNoodle commented:
If @mugling perhaps could take a look at it? As I did receive testing and feedback from the mod author, and it should be ready now.

I am generally wary of the risk of a pull request becoming old enough to provoke merge conflicts, but fortunately this being narrow in focus has avoided that.

BorkBorkGoesTheCode commented on data/mods/Arcana/monsters.json:
```diff
@@ -683,7 +683,7 @@
     "name": "archon",
     "species":"MUTANT",
-    "default_faction":"cult",
+    "default_faction":"archon",
```
You need to create an archon faction

BorkBorkGoesTheCode replied:
You need to create an archon faction

DangerNoodle replied:
Already resolved, see change to monster_factions.json.

DangerNoodle commented on data/mods/Arcana/monsters.json:
```diff
@@ -677,7 +677,7 @@
     "description":"A large plantlike creature with a gaping, toothy maw and numerous vines.  A hideous smell permeates the air around it.",
-    "flags":["HEARS", "SMELLS", "BASHES", "NOHEAD", "POISON", "NO_BREATHE", "PARALYZEVENOM"]
+    "flags":["HEARS", "SMELLS", "BASHES", "NOHEAD", "POISON", "NO_BREATHE", "PARALYZEVENOM", "PATH_AVOID_DANGER_2", "PRIORITIZE_TARGETS"]
     },{
```
Used by chicken walkers, police bots, eyebots, grackens, cleaner bots, mi-gos, riot control bots, Shia LaBeouf, tank drones, tribots, searchlights, zombie masters, zombie necromancers, and zombie scientists.

I am unsure what the full effects are, in detail. But it seems primarily used by intelligent monsters and robots. Of these, the archon is the one most likely intended to be intelligent. I added this to the moroboru in the hopes it will further mitigate any potential for infighting, but it may not be needed.

chaosvolt commented:
And from the looks of it, @Coolthulhu's second merging spree missed this yet again. Lovely. It's already been almost a month, and that's on top of the fact that this fixes the last remnants of bugs that are 3 or 4 months old (mainly, use action fuckery). -_-

illi-kun commented:
@chaosvolt your comment is annoying and useless. Who are you for manage developer's time? Did you pay salary to someone who need to answer to your expectation?

DangerNoodle commented:
While I can understand the potential frustration, it is not a helpful comment.

While I do hope it is merged, my only major concern is that the risk of a merge conflict developing increases with time, otherwise I can understand any delays.

mugling commented:
So there is or isn't a bug with diamond weapons as I don't see a fix for it in this PR? Neither do I see any changes involving `use_action`?

The remainder of this PR I'm not sure - none of the developers play or are familiar with the mod and there's a large queue of complex PR's open at the moment which are likely to take priority.

DangerNoodle commented:
There was previously an accidental bug caused by testing. I copied the mod folder from the repository to a compiled build. Because this folder was built off an older version, it threw up errors due to the diamond weapons that had been removed since the version of master that this pull request was based on. In other words, there is no actual bug outside of using old data in a new build.

As for use action, it is actually fixing an error introduced by Chaosvolt's fix for the hammer, as proposed in https://github.com/CleverRaven/Cataclysm-DDA/issues/18177. They presented two methods of fixing 'use_action' errors, and one of them led to unintended bugs.

DangerNoodle commented:
Thank you for the merge.
